app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java

```diff
@@ -44,9 +51,11 @@ public boolean onTouchEvent(MotionEvent event) {
         switch(event.getAction()) {
             case MotionEvent.ACTION_DOWN:
                 path.moveTo(pointX, pointY);
+                printTexturedCircle(pointX, pointY);
```

Comment:
You are creating a new circle each time you draw it.
Instead, draw it to a separate bitmap once, then draw the bitmap to canvasBitmap each time you want to draw the circle.